app/controllers/api/v1/gis_controller.rb

```diff
@@ -1,76 +1,83 @@
       address = geocoded_addresses[:candidates].first
-      match = address_within_neighborhood?(address)
+      proj_id = params[:project_id]
+      match = proj_id ? address_within_boundary?(address, proj_id) : nil
       return address.merge(boundary_match: match)
     else
       logger.error '<< GeocodingService.geocode candidates empty >> ' \
         "#{geocoded_addresses}, LOG PARAMS: #{log_params}"
-      notifications_sent = ArcGISNotificationService.new(
-        geocoded_addresses.merge(service_name: GeocodingService::NAME),
-        log_params,
-        params[:has_nrhp_adhp],
-      ).send_notifications
-
-      unless notifications_sent
-        log_msg = '<< GeocodingController.geocoding_data ' \
-          'send_notifications called but notifications not sent >>'
-        logger.error log_msg
-      end
+      send_geocoding_error_notification(geocoded_addresses)
 
       { boundary_match: nil }
     end
   end
 
-  def address_within_neighborhood?(address)
+  def address_within_boundary?(address, project_id)
+    return nil unless project_id.present? && address.present?
+
     x = address[:location][:x]
     y = address[:location][:y]
-    project_id = listing_params[:Project_ID]
-    return nil unless project_id.present?
     neighborhood = NeighborhoodBoundaryService.new(project_id, x, y)
     match = neighborhood.in_boundary?
-    # return successful geocoded data with the result of boundary_match
     matching_errors = neighborhood.errors
     return match unless matching_errors.present?
 
-    # otherwise notify of errors
     logger.error '<< NeighborhoodBoundaryService.in_boundary? Error >>' \
       "#{matching_errors}, LOG PARAMS: #{log_params}"
+    send_boundary_match_error_notification(matching_errors)
+
+    nil
+  end
+
+  def send_geocoding_error_notification(geocoded_addresses)
+    notifications_sent = ArcGISNotificationService.new(
+      geocoded_addresses.merge(service_name: GeocodingService::NAME),
+      log_params,
+    ).send_notifications
+
+    return if notifications_sent
+
+    log_msg = '<< GISController ' \
+      'send_geocoding_error_notification attempted but notifications not sent >>'
+    logger.error log_msg
+  end
+
+  def send_boundary_match_error_notification(matching_errors)
     notifications_sent = ArcGISNotificationService.new(
       {
-        errors: neighborhood.errors,
+        errors: matching_errors,
         service_name: NeighborhoodBoundaryService::NAME,
       },
       log_params,
-      params[:has_nrhp_adhp],
     ).send_notifications
 
-    return false if notifications_sent
-    log_msg = '<< GeocodingController.address_within_neighborhood? ' \
-      'send_notifications called but notifications not sent'
-    logger.error log_msg
+    return if notifications_sent
 
-    # default response
-    nil
+    log_msg = '<< GISController ' \
+      'send_boundary_match_error_notification attempted but notifications not sent >>'
+    logger.error log_msg
```

Comment:
@callaghanc Do we loose anything by not returning a `nil` response here?

Reply:
@creativecoder good q, here we are ok to not to have to return anything/worry about the return value. the `send_boundary_match_error_notification` and `send_geocoding_error_notification` methods only send the error notifications and do some logging. they don't handle/affect geocoding data or boundary match data so they don't have to return anything. we do have to make sure that `address_within_boundary?` returns `nil` when an address can't be boundary matched - see line 51 above. the notifications/logging stuff used to be fully mixed into the `geocoding_data` and `address_within_boundary?` methods, so before this PR we were sending notifications, doing logging, and then returning geocoding/boundary match data all in one method. this PR chops up the previously existing methods and does a bit of rearranging to break that up into more manageable chunks.